### ISSUE TITLE
Begin drawing more attention to vugu

### ISSUE THREAD

**Issue by ScreamingTaco** [vugu/vugu#53]
Note: I'm not sure if this is the right place to post this, but I couldn't think of anywhere else to do it.

## General Idea

Vugu seems like an excellent tool with a lot of potential. If we want it to grow faster and become more stable, we need to draw some attention to it from more developers. I had a few ideas (listed below). Does this sound like something more time should be invested in? What other ideas are there for spreading the word?

## Ideas

* Performance/size comparison with frameworks like vue or react
* Tutorials 
* Show how to implement something done in react, vue, angular, etc. in vugu

I'm pretty new to this project, but I'd be glad to help with some of these, if @vugu is interested.

**Comment by bradleypeabody** [comment]
Thanks very much for this - and overall yes, absolutely.  My concern is that there are still some important things to stabilize about the project before getting too deeply into detailed benchmarks, tutorials, etc.

One thing that would be useful is some sort of initial attempt at a "kitchen sink" app. As yet, Vugu doesn't have any sort UI library support (i.e. Vue has things like Vuetify as a framework of components - that doesn't exist yet for Vugu, it's too early).  But using something like either Bootstrap or MDL someone could try to build some forms and just generally use various UI components and see how it works out. I can guarantee at this stage that there will be un-figured-out, strange and buggy things that come up - but that's the point, with more real examples laid out on the table (so to speak) it will provide much needed experience to figure out the various loose ends.

If you want to take that on, by all means - let me know and we can chat about some of the trade offs and ideas of to approach it.

**Comment by ScreamingTaco** [comment]
Yeah I’d be interested in helping with that. I figure since I’m still learning vugu myself, I could start by doing shorter tutorials on smaller components, post them on my blog and medium, and then show how to make a more complete project once I’m capable of doing so. 

What would you consider a good starting point for the tutorials? I’m not super experienced with frontend stuff, but I was thinking on maybe finding a beginner vue tutorial and making the vugu equivalent.

Also, have you considered starting a vugu subreddit?

**Comment by bradleypeabody** [comment]
Cool, yeah that could work.  Although that's essentially what I was going for with the vugu site - https://www.vugu.org/doc/start is basically a small tutorial.   I'm curious what you run into when you try that (if you haven't already)? Maybe that experience can help guide it and give ideas about what's needed.

Making a vugu subreddit is definitely an idea.  I don't think I personally have the bandwidth to be on there very much answering people's questions at the moment.  I am on Slack however - https://invite.slack.golangbridge.org/, `#vugu` channel.

**Comment by ScreamingTaco** [comment]
The Vugu getting started tutorial is definitely helpful, but I think the more resources there are for a library/framework, the better. Someone who is intentional about giving vugu a try will read the docs and try to make something on their own, but I think guided tutorials appeal to a much broader audience. Anyway, I’ll get started and let you know how it goes. It may be a little bit before I’ve got it written and published, but I’ll keep you updated.

**Comment by bradleypeabody** [comment]
Cool, makes sense.  Yeah, definitely hit me with whatever questions, etc. as you go.

**Comment by ScreamingTaco** [comment]
First tutorial: https://screamingtaco.github.io/post/2019-08-26-vugu-tutorial-prt1/

I’ll be seeing about cross posting to medium soon, but I can’t make any guarantees.

**Comment by bradleypeabody** [comment]
Thanks @ScreamingTaco, this is great! 

Some of the things you got tripped up on (e.g. Root vs RootData and where to put your methods), will be changing as soon as I can merge in the component-refactor branch - and I'll update the docs for that too.   It might be worth doing a new updated version of this article once that is available.

I'll hit you up on slack for further discussion.

**Comment by ScreamingTaco** [comment]
Sounds good 👍

**Comment by bradleypeabody** [comment]
No problem 👍.  This was the message from Slack: 

> Thanks for the tutorial post!  https://screamingtaco.github.io/post/2019-08-26-vugu-tutorial-prt1/  Some of the key things that you ran into are addressed on the component-refactor branch, which I hope to be able to merge into master next week.  (Things like instead of having Root and RootData - each component will just be one struct.  And when you get an event relevant DOM data will be readily available on the DOMEvent struct, without having to do syscall/js calls.)
